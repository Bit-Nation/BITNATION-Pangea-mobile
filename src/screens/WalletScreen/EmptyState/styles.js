import { MediaQueryStyleSheet } from 'react-native-responsive';

import GlobalStyles from '../../../global/Styles';

export default styles = MediaQueryStyleSheet.create({
  ...GlobalStyles,
  card: {
    marginLeft: 8,
    marginRight: 8,
  },
  cardContainer: {
    flex: 1,
    justifyContent: 'flex-start',
  },
  firstCard: {
    marginBottom: 8,
  },
});
  