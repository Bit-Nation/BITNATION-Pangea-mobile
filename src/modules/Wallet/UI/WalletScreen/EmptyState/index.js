// @flow

import React from 'react';
import { View, ScrollView } from 'react-native';

import styles from './styles';
import PanelView from '../../../components/common/PanelView';
import i18n from '../../../global/i18n';

type Props = {
  /**
   * @desc Callback when create wallet action selected.
   */
  onCreateWallet: () => void,
  /**
   * @desc Callback when restore wallet action selected.
   */
  onRestoreWallet: () => void,
}

/**
 * @desc Component that renders wallet screen empty state.
 * It is embedded into parent screen, so doesn't contain navigation, etc.
 * @return {React.Component} A component.
 * @constructor
 */
const EmptyWalletScreen = ({ onCreateWallet, onRestoreWallet }: Props) => ((
  <View>
    <ScrollView style={styles.scrollViewWallet}>
      <PanelView
        style={styles.panelViewTransparent}
        title={i18n.t('screens.wallet.empty.createWalletPanel.title')}
        body={i18n.t('screens.wallet.empty.createWalletPanel.text')}
        buttonTitle={i18n.t('screens.wallet.empty.createWalletPanel.button')}
        onButtonClick={onCreateWallet}
      />
      <PanelView
        style={styles.panelViewTransparent}
        title={i18n.t('screens.wallet.empty.restoreWalletPanel.title')}
        body={i18n.t('screens.wallet.empty.restoreWalletPanel.text')}
        buttonTitle={i18n.t('screens.wallet.empty.restoreWalletPanel.button')}
        onButtonClick={onRestoreWallet}
      />
    </ScrollView>
  </View>
));

export default EmptyWalletScreen;
