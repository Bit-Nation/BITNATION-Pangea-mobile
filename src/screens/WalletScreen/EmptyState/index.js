import React, { Component } from 'react';
import {
  ScrollView,
  View,
} from 'react-native';
import PropTypes from 'prop-types';

import styles from './styles';
import BackgroundImage from '../../../components/common/BackgroundImage';
import MessageView from '../../../components/common/MessageView';
import FakeNavigationBar from '../../../components/common/FakeNavigationBar';

export default class EmptyWalletScreen extends Component {

  texts = {
    createWalletTitle: 'Create a Wallet',
    createWalletText: 'A wallet holds money, such as Ethereum. You will need a wallet to pay for services, such as creating and editing nations, notarizing documents, and other useful services.',
    restoreWalletTitle: 'Restore a Wallet',
    restoreWalletText: 'If you need to restore your wallet (because you lost or upgraded your phone) use the button below.\nYou will need your Paper Key to restore your wallet.',
  };

  render() {
    return (
      <View style={styles.container}>
        <BackgroundImage/>
        <FakeNavigationBar/>
        <ScrollView>
          <View style={styles.cardContainer}>
            <View style={[styles.card, styles.firstCard]}>
              <MessageView
                title={this.texts.createWalletTitle}
                messageText={this.texts.createWalletText}
                buttonTitle='Create a Wallet'
                onButtonClick={this.props.onCreateWallet}
              />
            </View>

            <View style={styles.card}>
              <MessageView
                title={this.texts.restoreWalletTitle}
                messageText={this.texts.restoreWalletText}
                buttonTitle='Restore a Wallet'
                onButtonClick={this.props.onRestoreWallet}
              />
            </View>
          </View>
        </ScrollView>
      </View>
    );
  }
}

EmptyWalletScreen.PropTypes = {
  onCreateWallet: PropTypes.func.isRequired,
  onRestoreWallet: PropTypes.func.isRequired,
};

