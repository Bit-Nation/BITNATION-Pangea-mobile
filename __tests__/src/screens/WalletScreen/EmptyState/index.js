import React from 'react';
import { shallow } from 'enzyme';

import EmptyState from '../../../../../src/screens/WalletScreen/EmptyState';

test('EmptyState renders correctly', () => {
  const wrapper = shallow((
    <EmptyState />
  ));
  expect(wrapper).toMatchSnapshot();
});
