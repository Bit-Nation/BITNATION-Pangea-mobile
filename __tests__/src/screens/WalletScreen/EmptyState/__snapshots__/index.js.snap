// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmptyWalletScreen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <ScrollViewMock
        style={
          Object {
            "flex": 0,
            "marginBottom": 56,
          }
        }
      >
        <PanelView
          body="You need an Ethereum wallet to make payments, to create nations, and to use other useful services."
          buttonTitle="Create Wallet"
          childrenContainerStyle={Object {}}
          icon={undefined}
          onButtonClick={null}
          renderAdditionalInfo={null}
          renderBottom={null}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 8,
              "flexDirection": "column",
              "justifyContent": "space-between",
              "marginBottom": 8,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 8,
              "paddingBottom": 4,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 4,
            }
          }
          title="Create Wallet"
          titleStyle={
            Object {
              "color": "#58595B",
              "fontFamily": "Roboto",
              "fontSize": 22,
              "fontWeight": "bold",
              "letterSpacing": 0,
              "lineHeight": 28,
              "textAlign": "left",
            }
          }
        />
        <PanelView
          body="You can restore your wallet if you lose or change your phone. You will need your private key to restore your wallet."
          buttonTitle="Restore Wallet"
          childrenContainerStyle={Object {}}
          icon={undefined}
          onButtonClick={null}
          renderAdditionalInfo={null}
          renderBottom={null}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 8,
              "flexDirection": "column",
              "justifyContent": "space-between",
              "marginBottom": 8,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 8,
              "paddingBottom": 4,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 4,
            }
          }
          title="Restore Wallet"
          titleStyle={
            Object {
              "color": "#58595B",
              "fontFamily": "Roboto",
              "fontSize": 22,
              "fontWeight": "bold",
              "letterSpacing": 0,
              "lineHeight": 28,
              "textAlign": "left",
            }
          }
        />
      </ScrollViewMock>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <PanelView
            body="You need an Ethereum wallet to make payments, to create nations, and to use other useful services."
            buttonTitle="Create Wallet"
            childrenContainerStyle={Object {}}
            icon={undefined}
            onButtonClick={null}
            renderAdditionalInfo={null}
            renderBottom={null}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderRadius": 8,
                "flexDirection": "column",
                "justifyContent": "space-between",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 8,
                "paddingBottom": 4,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 4,
              }
            }
            title="Create Wallet"
            titleStyle={
              Object {
                "color": "#58595B",
                "fontFamily": "Roboto",
                "fontSize": 22,
                "fontWeight": "bold",
                "letterSpacing": 0,
                "lineHeight": 28,
                "textAlign": "left",
              }
            }
          />,
          <PanelView
            body="You can restore your wallet if you lose or change your phone. You will need your private key to restore your wallet."
            buttonTitle="Restore Wallet"
            childrenContainerStyle={Object {}}
            icon={undefined}
            onButtonClick={null}
            renderAdditionalInfo={null}
            renderBottom={null}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderRadius": 8,
                "flexDirection": "column",
                "justifyContent": "space-between",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 8,
                "paddingBottom": 4,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 4,
              }
            }
            title="Restore Wallet"
            titleStyle={
              Object {
                "color": "#58595B",
                "fontFamily": "Roboto",
                "fontSize": 22,
                "fontWeight": "bold",
                "letterSpacing": 0,
                "lineHeight": 28,
                "textAlign": "left",
              }
            }
          />,
        ],
        "style": Object {
          "flex": 0,
          "marginBottom": 56,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "body": "You need an Ethereum wallet to make payments, to create nations, and to use other useful services.",
            "buttonTitle": "Create Wallet",
            "children": null,
            "childrenContainerStyle": Object {},
            "icon": undefined,
            "onButtonClick": null,
            "renderAdditionalInfo": null,
            "renderBottom": null,
            "style": Object {
              "backgroundColor": "transparent",
              "borderRadius": 8,
              "flexDirection": "column",
              "justifyContent": "space-between",
              "marginBottom": 8,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 8,
              "paddingBottom": 4,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 4,
            },
            "title": "Create Wallet",
            "titleStyle": Object {
              "color": "#58595B",
              "fontFamily": "Roboto",
              "fontSize": 22,
              "fontWeight": "bold",
              "letterSpacing": 0,
              "lineHeight": 28,
              "textAlign": "left",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "body": "You can restore your wallet if you lose or change your phone. You will need your private key to restore your wallet.",
            "buttonTitle": "Restore Wallet",
            "children": null,
            "childrenContainerStyle": Object {},
            "icon": undefined,
            "onButtonClick": null,
            "renderAdditionalInfo": null,
            "renderBottom": null,
            "style": Object {
              "backgroundColor": "transparent",
              "borderRadius": 8,
              "flexDirection": "column",
              "justifyContent": "space-between",
              "marginBottom": 8,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 8,
              "paddingBottom": 4,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 4,
            },
            "title": "Restore Wallet",
            "titleStyle": Object {
              "color": "#58595B",
              "fontFamily": "Roboto",
              "fontSize": 22,
              "fontWeight": "bold",
              "letterSpacing": 0,
              "lineHeight": 28,
              "textAlign": "left",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <ScrollViewMock
          style={
            Object {
              "flex": 0,
              "marginBottom": 56,
            }
          }
        >
          <PanelView
            body="You need an Ethereum wallet to make payments, to create nations, and to use other useful services."
            buttonTitle="Create Wallet"
            childrenContainerStyle={Object {}}
            icon={undefined}
            onButtonClick={null}
            renderAdditionalInfo={null}
            renderBottom={null}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderRadius": 8,
                "flexDirection": "column",
                "justifyContent": "space-between",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 8,
                "paddingBottom": 4,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 4,
              }
            }
            title="Create Wallet"
            titleStyle={
              Object {
                "color": "#58595B",
                "fontFamily": "Roboto",
                "fontSize": 22,
                "fontWeight": "bold",
                "letterSpacing": 0,
                "lineHeight": 28,
                "textAlign": "left",
              }
            }
          />
          <PanelView
            body="You can restore your wallet if you lose or change your phone. You will need your private key to restore your wallet."
            buttonTitle="Restore Wallet"
            childrenContainerStyle={Object {}}
            icon={undefined}
            onButtonClick={null}
            renderAdditionalInfo={null}
            renderBottom={null}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderRadius": 8,
                "flexDirection": "column",
                "justifyContent": "space-between",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 8,
                "paddingBottom": 4,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 4,
              }
            }
            title="Restore Wallet"
            titleStyle={
              Object {
                "color": "#58595B",
                "fontFamily": "Roboto",
                "fontSize": 22,
                "fontWeight": "bold",
                "letterSpacing": 0,
                "lineHeight": 28,
                "textAlign": "left",
              }
            }
          />
        </ScrollViewMock>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <PanelView
              body="You need an Ethereum wallet to make payments, to create nations, and to use other useful services."
              buttonTitle="Create Wallet"
              childrenContainerStyle={Object {}}
              icon={undefined}
              onButtonClick={null}
              renderAdditionalInfo={null}
              renderBottom={null}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 8,
                  "flexDirection": "column",
                  "justifyContent": "space-between",
                  "marginBottom": 8,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 8,
                  "paddingBottom": 4,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 4,
                }
              }
              title="Create Wallet"
              titleStyle={
                Object {
                  "color": "#58595B",
                  "fontFamily": "Roboto",
                  "fontSize": 22,
                  "fontWeight": "bold",
                  "letterSpacing": 0,
                  "lineHeight": 28,
                  "textAlign": "left",
                }
              }
            />,
            <PanelView
              body="You can restore your wallet if you lose or change your phone. You will need your private key to restore your wallet."
              buttonTitle="Restore Wallet"
              childrenContainerStyle={Object {}}
              icon={undefined}
              onButtonClick={null}
              renderAdditionalInfo={null}
              renderBottom={null}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderRadius": 8,
                  "flexDirection": "column",
                  "justifyContent": "space-between",
                  "marginBottom": 8,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 8,
                  "paddingBottom": 4,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 4,
                }
              }
              title="Restore Wallet"
              titleStyle={
                Object {
                  "color": "#58595B",
                  "fontFamily": "Roboto",
                  "fontSize": 22,
                  "fontWeight": "bold",
                  "letterSpacing": 0,
                  "lineHeight": 28,
                  "textAlign": "left",
                }
              }
            />,
          ],
          "style": Object {
            "flex": 0,
            "marginBottom": 56,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "body": "You need an Ethereum wallet to make payments, to create nations, and to use other useful services.",
              "buttonTitle": "Create Wallet",
              "children": null,
              "childrenContainerStyle": Object {},
              "icon": undefined,
              "onButtonClick": null,
              "renderAdditionalInfo": null,
              "renderBottom": null,
              "style": Object {
                "backgroundColor": "transparent",
                "borderRadius": 8,
                "flexDirection": "column",
                "justifyContent": "space-between",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 8,
                "paddingBottom": 4,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 4,
              },
              "title": "Create Wallet",
              "titleStyle": Object {
                "color": "#58595B",
                "fontFamily": "Roboto",
                "fontSize": 22,
                "fontWeight": "bold",
                "letterSpacing": 0,
                "lineHeight": 28,
                "textAlign": "left",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "body": "You can restore your wallet if you lose or change your phone. You will need your private key to restore your wallet.",
              "buttonTitle": "Restore Wallet",
              "children": null,
              "childrenContainerStyle": Object {},
              "icon": undefined,
              "onButtonClick": null,
              "renderAdditionalInfo": null,
              "renderBottom": null,
              "style": Object {
                "backgroundColor": "transparent",
                "borderRadius": 8,
                "flexDirection": "column",
                "justifyContent": "space-between",
                "marginBottom": 8,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 8,
                "paddingBottom": 4,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 4,
              },
              "title": "Restore Wallet",
              "titleStyle": Object {
                "color": "#58595B",
                "fontFamily": "Roboto",
                "fontSize": 22,
                "fontWeight": "bold",
                "letterSpacing": 0,
                "lineHeight": 28,
                "textAlign": "left",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
